### PR TITLE
Add to preselected rights

### PR DESCRIPTION
**Pull request by pldespaigne**
tag issue #5706

_moved from #5826_

### CODE REVIEW

**Review by RemcoSimonides — APPROVED** [review]
LGTM

File: libs/contract/src/lib/avails/avails.ts [hunk 1]
```diff
@@ -144,8 +144,45 @@ export function isInBucket(avails: AvailsFilter, terms: BucketTerm[]) {
...
+  if (term.exclusive !== avail.exclusive) return false;
+
+  if (
+    !avail.medias
+    || term.medias.length !== avail.medias.length
+    || term.medias.some(medium => !avail.medias.includes(medium))
+  ) return false;
+
+  return true;
+}
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.ts:165]:
```typescript
export function isSameMediaTerm(term: BucketTerm, avail: AvailsFilter) {
  if (!avail.medias) return false;
  if (term.medias.length !== avail.medias.length) return false;
  if (term.medias.some(medium => !avail.medias.includes(medium))) return false;
  return true;
}
```

File: libs/contract/src/lib/avails/avails.ts [hunk 2]
```diff
@@ -144,8 +144,45 @@ export function isInBucket(avails: AvailsFilter, terms: BucketTerm[]) {
 ///////////
 // utils //
 ///////////
-export function findSameTermIndex(terms: BucketTerm[], avail: AvailsFilter) {
-  return terms.findIndex(t => isSameTerm(t, avail));
+export function findSameTerritoriesTermIndex(terms: BucketTerm[], avail: AvailsFilter) {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.ts:147]:
```typescript
export function findSameTerritoriesTermIndex(terms: BucketTerm[], avail: AvailsFilter) {
  const isSameTerm = (term) => {
    if (term.exclusive !== avail.exclusive) return false;
    return isSameMediaTerm(avail, term) && isSameTerritoriesTerm(avail, term);
  }
  return terms.findIndex(isSameTerm);
}
```

File: libs/contract/src/lib/bucket/form.ts [hunk 3]
```diff
@@ -189,9 +189,51 @@ export class BucketForm extends FormEntity<BucketControls, Bucket> {
...
     if (contractIndex === -1) { return false; }
     const contract = bucket.contracts[contractIndex];
-    const termIndex = findSameTermIndex(contract.terms, avails);
+    const termIndex = findSameTerritoriesTermIndex(contract.terms, avails);
     if (termIndex === -1) { return false }
     const territories = contract.terms[termIndex].territories;
     return territories.includes(territory);
   }
+
+  addDuration(avails: AvailsFilter, marker: DurationMarker) {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/bucket/form.ts:198]: Add comment to same "use in the calendar"

File: libs/contract/src/lib/avails/avails.ts [hunk 4]
```diff
@@ -144,8 +144,45 @@ export function isInBucket(avails: AvailsFilter, terms: BucketTerm[]) {
...
+  return terms.findIndex(t => isSameTerritoriesTerm(t, avail) && isSameMediaTerm(t, avail));
+}
+
+export function findSameDurationTermIndex(terms: BucketTerm[], avail: AvailsFilter) {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.ts:151]:
```typescript
export function findSameDurationTermIndex(terms: BucketTerm[], avail: AvailsFilter) {
  const isSameTerm = (term) => {
    if (term.exclusive !== avail.exclusive) return false;
    return isSameMediaTerm(avail, term) && isSameDurationTerm(avail, term);
  }
  return terms.findIndex(isSameTerm);
}
```

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**